### PR TITLE
Add HTTP status code to iModels error

### PR DESCRIPTION
**Pull request by arminasbrazenas**
- Add HTTP status code to iModels error
- Add `MutuallyExclusiveParametersProvided` as known error for `$search` parameter

### CODE REVIEW

File: clients/imodels-client-management/src/base/internal/IModelsErrorParser.ts [hunk 1]
```diff
@@ -36,10 +37,12 @@ export class IModelsErrorBaseImpl extends Error {
 
 export class IModelsErrorImpl extends IModelsErrorBaseImpl implements IModelsError {
   public details?: IModelsErrorDetail[];
+  public statusCode?: number;
 
-  constructor(params: { code: IModelsErrorCode, message: string, details?: IModelsErrorDetail[] }) {
+  constructor(params: { code: IModelsErrorCode, message: string, details?: IModelsErrorDetail[], statusCode?: number }) {
```
Comment (austeja-bentley) [inline clients/imodels-client-management/src/base/internal/IModelsErrorParser.ts:42]: could we make `statusCode` and `details` mandatory params? Having `statusCode` as an optional argument makes it easier for us to leave a bug and not pass it to constructor.

Comment (arminasbrazenas) [reply clients/imodels-client-management/src/base/internal/IModelsErrorParser.ts:42]: Mandatory as passing 'undefined' explicitly?

Comment (austeja-bentley) [reply clients/imodels-client-management/src/base/internal/IModelsErrorParser.ts:42]: yes

Comment (arminasbrazenas) [reply clients/imodels-client-management/src/base/internal/IModelsErrorParser.ts]: updated

File: clients/imodels-client-management/src/base/internal/AxiosRestClient.ts [hunk 2]
```diff
@@ -7,12 +7,13 @@ import axios, { AxiosRequestConfig, AxiosResponse } from "axios";
...
 
 import { AxiosResponseHeadersAdapter } from "./AxiosResponseHeadersAdapter";
+import { ResponseInfo } from "./IModelsErrorParser";
 
 /**
  * Function that is called if the HTTP request fails and which returns an error that will be thrown by one of the
  * methods in {@link RestClient}.
  */
-export type ParseErrorFunc = (response: { statusCode?: number, body?: unknown }, originalError: Error & { code?: string }) => Error;
+export type ParseErrorFunc = (response: ResponseInfo, originalError: Error & { code?: string }) => Error;
```
[the remark this answers is not in the document]
Comment (arminasbrazenas) [reply clients/imodels-client-management/src/base/internal/AxiosRestClient.ts:16]: it's in internal folder, should it be used by package consumers?

**Review by austeja-bentley — APPROVED** [review]

**Pull request merged**